RadixWallet/Clients/DeviceFactorSourceClient/DeviceFactorSourceClient+Live.swift

```diff
@@ -15,61 +15,22 @@ extension DeviceFactorSourceClient: DependencyKey {
 
 		let entitiesControlledByFactorSource: GetEntitiesControlledByFactorSource = { factorSource, maybeSnapshot in
-
-			let (allNonHiddenEntities, allHiddenEntities) = try await { () -> (allNonHiddenEntities: [AccountOrPersona], allHiddenEntities: [AccountOrPersona]) in
-				let accountNonHidden: [Account]
-				let accountHidden: [Account]
-				let personasNonHidden: [Persona]
-				let personasHidden: [Persona]
-
-				if let overridingSnapshot = maybeSnapshot {
-					let networkID = NetworkID.mainnet
-					let profile = overridingSnapshot
-					let network = try profile.network(id: networkID)
-					accountNonHidden = network.getAccounts().elements
-					personasNonHidden = network.getPersonas().elements
-
-					accountHidden = network.getHiddenAccounts().elements
-					personasHidden = network.getHiddenPersonas().elements
-				} else {
-					accountNonHidden = try await accountsClient.getAccountsOnCurrentNetwork().elements
-					personasNonHidden = try await personasClient.getPersonas().elements
-
-					accountHidden = try await accountsClient.getHiddenAccountsOnCurrentNetwork().elements
-					personasHidden = try await personasClient.getHiddenPersonasOnCurrentNetwork().elements
-				}
-
-				var allNonHiddenEntities = accountNonHidden.map(AccountOrPersona.account)
-				allNonHiddenEntities.append(contentsOf: personasNonHidden.map(AccountOrPersona.persona))
-
-				var allHidden = accountHidden.map(AccountOrPersona.account)
-				allHidden.append(contentsOf: personasHidden.map(AccountOrPersona.persona))
-
-				return (allNonHiddenEntities, allHidden)
-			}()
-
-			let nonHiddenEntitiesForSource = allNonHiddenEntities.filter { entity in
-				switch entity.securityState {
-				case let .unsecured(unsecuredEntityControl):
-					unsecuredEntityControl.transactionSigning.factorSourceID == factorSource.id
-				}
+			let profileToCheck: ProfileToCheck = if let maybeSnapshot {
+				.specific(maybeSnapshot)
+			} else {
+				.current
 			}
-
-			let hiddenEntitiesForSource = allHiddenEntities.filter { entity in
-				switch entity.securityState {
-				case let .unsecured(unsecuredEntityControl):
-					unsecuredEntityControl.transactionSigning.factorSourceID == factorSource.id
-				}
+			let result = try await SargonOS.shared.entitiesLinkedToFactorSource(factorSource: factorSource.asGeneral, profileToCheck: profileToCheck)
+			guard case let .device(integrity) = result.integrity else {
+				struct UnexpectedFactorSource: Error {}
+				throw UnexpectedFactorSource()
 			}
-
-			let isMnemonicMarkedAsBackedUp = userDefaults.getFactorSourceIDOfBackedUpMnemonics().contains(factorSource.id)
-
 			return EntitiesControlledByFactorSource(
-				entities: nonHiddenEntitiesForSource,
-				hiddenEntities: hiddenEntitiesForSource,
+				entities: result.accounts.map(AccountOrPersona.account) + result.personas.map(AccountOrPersona.persona),
+				hiddenEntities: result.hiddenAccounts.map(AccountOrPersona.account) + result.hiddenPersonas.map(AccountOrPersona.persona),
```

Comment:
feels like we want `result.hiddenAccounts.map(AccountOrPersona.account) + result.hiddenPersonas.map(AccountOrPersona.persona),` in Sargon...?

Reply:
and `result.accounts.map(AccountOrPersona.account) + result.personas.map(AccountOrPersona.persona),` too

Reply:
not really, since it is only for this particular case of entities linked to `DeviceFactorSource` that we want to treat it as `AccountOrPersona` lists. In fact, we may even end up updating the usages of this method (could be using the new UI for other flows) and end removing this implementation as well (and always use `SargonOS` output, as we do for new `FactorSourcesList`).